beacon-chain/state/stategen/valcounter.go

```diff
@@ -0,0 +1,68 @@
+package stategen
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/core/helpers"
+	coreTime "github.com/prysmaticlabs/prysm/v3/beacon-chain/core/time"
+	beacondb "github.com/prysmaticlabs/prysm/v3/beacon-chain/db"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	"github.com/prysmaticlabs/prysm/v3/encoding/bytesutil"
+)
+
+type ActiveValidatorCounter interface {
+	ActiveValidatorCount(ctx context.Context) (uint64, error)
+}
+
+type LastFinalizedValidatorCounter struct {
+	count uint64
+	db    beacondb.HeadAccessDatabase
+	sg    StateManager
```

Comment:
`sg` for `stategen`, `sm` for `StateManager` - am I missing something?

Reply:
I just meant that they are nearly synonymous in my mind, as `stategen.State` is basically the stategen service - `StateManager` seems like it exists to mirror the stategen service so it can be mocked etc, and I usually use the concrete type not the interface, so it's muscle memory, I'll change it :)